compiler/rustc_passes/Cargo.toml

```diff
@@ -5,6 +5,7 @@ edition = "2021"
 
 [dependencies]
 tracing = "0.1"
+itertools = "0.10.1"
```

Comment:
I'm failing to see where itertools is being used now, is still necessary after the previous rounds of reviews?

Reply:
`group_by` in `itertools` is used for grouping dead_codes by their log levels.

ref: https://github.com/rust-lang/rust/pull/97853/files#r898454245